From 6283c7ba907cef3fbbabdf38171c2709c69ebaea Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Fri, 9 Mar 2018 07:59:58 +0100
Subject: [PATCH] perfetto_cmd: Resolve missing O_CREAT mode

 src/perfetto_cmd/perfetto_cmd.cc:359:65:
  error: 'open' has superfluous mode bits; missing O_CREAT?

Change-Id: Id051a878a93268c42cd4d3e664247b98ed40dc25
---
 src/perfetto_cmd/perfetto_cmd.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/perfetto_cmd/perfetto_cmd.cc b/src/perfetto_cmd/perfetto_cmd.cc
index 9614642c..ebdd740b 100644
--- a/src/perfetto_cmd/perfetto_cmd.cc
+++ b/src/perfetto_cmd/perfetto_cmd.cc
@@ -356,7 +356,7 @@ bool PerfettoCmd::OpenOutputFile() {
     // If we are tracing to DropBox, there's no need to make a
     // filesystem-visible temporary file.
     // TODO(skyostil): Fall back to base::TempFile for older devices.
-    fd.reset(open(kTempDropBoxTraceDir, O_TMPFILE | O_RDWR, 0600));
+    fd.reset(open(kTempDropBoxTraceDir, O_CREAT | O_TMPFILE | O_RDWR, 0600));
     if (!fd) {
       PERFETTO_ELOG("Could not create a temporary trace file in %s",
                     kTempDropBoxTraceDir);
-- 
2.17.1

